src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MonoDroidMarkStep.cs

```diff
@@ -436,6 +436,14 @@ protected override void DoAdditionalMethodProcessing (MethodDefinition method)
 			PreserveRegisteredMethod (method.DeclaringType, member);
 		}
 
+		protected override bool ShouldMarkInterfaceImplementation (TypeDefinition type, InterfaceImplementation iface, TypeDefinition resolvedInterfaceType)
+		{
+			if (base.ShouldMarkInterfaceImplementation (type, iface, resolvedInterfaceType))
+				return true;
+
+			return resolvedInterfaceType.TryGetRegisterAdapter (out _);
```

Comment:
Based on [these musings](https://github.com/xamarin/xamarin-android/pull/3525#issuecomment-523270275), I don't think that this method should look for `[Register]` on an interface to determine if the interface type should be preserved.  The 3rd constructor argument is meaningless, and ~anybody can apply `[Register]` on their type.  (Why they would *do* so, ¯\_(ツ)_/¯)

```csharp
[Register("terrible/Example", "", "I mean really terrible, no?")]
public interface IBadExample {
}
```

Instead, I think a *better* indicator would be if `resolvedInterfaceType` implements the `IJavaPeerable` or `IJavaObject` interfaces.  Yes, this can also be done by anyone, but it's also *semantically meaningful*: if the interface implements `IJavaPeerable`/`IJavaObject`, then it can be passed into Java code and gains lots of special abilities (via `Java.Lang.Object` and JCW generation and...):

```csharp
[Register("better/Example", "", "I still don't think that this value is meaningfully used")]
public interface IBetterExample : IJavaObject, IJavaPeerable {
}
```

Reply:
That would work too. I wanted to make it close to what linker did before (https://github.com/xamarin/xamarin-android/commit/5b945abb2e1f0fe1a9ab6589993c970869c4a752), thus I reused the code, which used and checked the Register attribute on the iface.

Wonder about the Register attribute on types though. Why do we still generate it with the 3rd argument then? Was the linker only user of it? Do we have documentation for the Register attributes our generator produces?

Reply:
> Why do we still generate it with the 3rd argument then?

I think it was mostly a "this makes sense" decision, rather than anything concrete, as it (kinda) "mirrored" the connector value for `[Register]` on methods, which *is* used.

I can't find any evidence of the 3rd parameter ever being used on *types*, *certainly* not to find the `Invoker` for the interface (which is determined by appending `Invoker` to the type name, as linked elsewhere).

Thus, I think we should *stop* emitting that 3rd argument within `generator`.

The only reason I can think of to keep it is to use that value preferentially over appending `Invoker` to the type name.  We should measure this, but I doubt that a `Type.GetCustomAttributes(typeof(RegisterAttribute))` invocation will be faster than the existing "grab the type name and append `Invoker`" approach.

(Additionally, with the introduction of C#8 default interface methods, I'm highly tempted to just require that the `Invoker` type be a nested type, which could be done consistently for *everything*, and would instead involve a `Type.GetNestedType()` invocation.  However, this would probably constitute an ABI break, as newer assemblies couldn't work with previous releases...)

> Do we have documentation for the Register attributes our generator produces

Somewhat: https://docs.microsoft.com/en-us/xamarin/android/platform/java-integration/working-with-jni#binding-interfaces

I don't find any uses of `[Register]` *on types* within that document in which more than one constructor parameter is provided.